pkg/api/authz/mcpserver.go

```diff
@@ -44,7 +44,7 @@ func (a *Authorizer) checkMCPServer(req *http.Request, resources *Resources, u u
 
 		resources.Authorizated.MCPServer = &mcpServer
 		return true, nil
-	} else if mcpServer.Spec.MCPServerCatalogEntryName != "" {
+	} else if mcpServer.Spec.MCPServerCatalogEntryName != "" && resources.Authorizated.PowerUserWorkspace != nil {
```

Comment:
Two lines down, we deref `resources.Authorizated.PowerUserWorkspace`, assuming it is non-nil.